### PR TITLE
Remove dependency on hiveclient.

### PR DESCRIPTION
**Pull request by xuezhaojun**
Fixes: https://issues.redhat.com/browse/ACM-11089

### CODE REVIEW

File: test/e2e/util/hive.go [hunk 1]
```diff
@@ -194,11 +222,21 @@ func CleanClusterPool(hiveClient hiveclient.Interface, name, namespace string) e
 	return DeleteNamespace(namespace)
 }
 
-func ClaimCluster(hiveClient hiveclient.Interface, cdName, cdNs, claimName string) error {
-	cd, err := hiveClient.HiveV1().ClusterDeployments(cdNs).Get(context.TODO(), cdName, metav1.GetOptions{})
+func ClaimCluster(hiveClient client.Client, cdName, cdNs, claimName string) error {
```
Comment (xuezhaojun) [inline test/e2e/util/hive.go:225]: This function contains the only logic change in this PR, from the name and usage I see it attempt to add the clusterclaim to cluster, so I added the Update part to it. @zhiweiyin318 

cc @ldpliu

**Comment by ldpliu** [comment]
it looks good to me.
/unhold

**Pull request merged**